src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -559,13 +560,39 @@ private static Details getDetails(DetailAST ast) {
                 case TokenTypes.LITERAL_DO:
                     details = getDetailsForDoLoops(ast);
                     break;
+                case TokenTypes.LITERAL_SWITCH:
+                    details = getDetailsForSwitch(ast);
+                    break;
                 default:
                     details = getDetailsForOthers(ast);
                     break;
             }
             return details;
         }
 
+        /**
+         * Collect all the requirement to check violation for switch statements and expressions.
+         *
+         * @param ast Node which pointing to Token Switch case
```

Comment:
switch statement or expression to gather details about

Reply:
done